rfcs/0084-ZHF-on-master-and-freeze-dates.md

```diff
@@ -0,0 +1,104 @@
+---
+feature: nixos-freeze-dates
+start-date: 2021-01-17
+author: Jonathan Ringer (@jonringer)
+co-authors:
+related-issues: [NixOS release schedule](https://github.com/NixOS/rfcs/pull/80)
+---
+
+# Summary
+[summary]: #summary
+
+To further minimize the amount of risk going into a NixOS release, I propose
+to set freeze dates regarding changes to critical packages being merged to
+staging. In addition, I also propose to stabilize the release on master
+instead of the release branch, thus avoiding the need to backport every pull
+request, and take advantage of the stabilization that unstable normally
+receives.
+
+# Definitions
+[definitions]: #definition
+
+- "Freezing" a branch or package.
+  - Disallow breaking changes to branch or package. In SemVer terminology, disallow major version bumps.
```

Comment:
I'm wondering if we should define an exception here: breaking changes can be allowed if those are necessary to fix security issues or fix a critical bug (i.e. something that corrupts your data or crashes your system).

Note: while backporting single patches is the preferred way in that situation, those can be "breaking" as well.

Reply:
Yea, and that's the exception I make for the release channel. But only if patching isn't a reasonable alternative.